lighthouse-core/index.js

```diff
@@ -18,27 +20,30 @@ const Config = require('./config/config');
  *   runner.js - marshalls the actions that must be taken (Gather / Audit)
  *               config file is used to determine which of these actions are needed
  *
- *   lighthouse-cli \
- *                   -- index.js  \
- *                                 ----- runner.js ----> [Gather / Audit]
- *           lighthouse-extension /
- *
+ *         lighthouse-cli \
+ *                         -- core/index.js ----> runner.js ----> [Gather / Audit]
+ *   lighthouse-extension /
  */
 
 /**
- * @param {string=} url
- * @param {LH.Flags=} flags
- * @param {LH.Config.Json=} configJSON
+ * Run Lighthouse.
+ * @param {string=} url The URL to test. Optional if running in auditMode.
+ * @param {LH.Flags=} flags Optional settings for the Lighthouse run. If present,
+ *   they will override any settings in the config.
+ * @param {LH.Config.Json=} configJSON Configuration for the Lighthouse run. If
+ *   not present, the default config is used.
+ * @param {Connection=} connection
  * @return {Promise<LH.RunnerResult|undefined>}
  */
-async function lighthouse(url, flags = {}, configJSON) {
+async function lighthouse(url, flags = {}, configJSON, connection) {
```

Comment:
if we feel like this is going to become a never-ending list of optional arguments, we could make this an object (`overrides`? or...something)

Reply:
4️⃣. 0️⃣  😄

Reply:
> 4️⃣. 0️⃣

ha. Just to be clear, ambiguous smiley face, this should be fine since it's just an addition to our API, not a change, right?